azext_iot/dps/commands_device_registration.py

```diff
@@ -0,0 +1,48 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+
+from knack.log import get_logger
+
+from azext_iot.dps.providers.device_registration import DeviceRegistrationProvider
+
+logger = get_logger(__name__)
+
+
+def create_device_registration(
+    cmd,
+    registration_id: str,
+    enrollment_group_id: str = None,
+    device_symmetric_key: str = None,
+    compute_key: bool = False,
+    certificate_file: str = None,
+    key_file: str = None,
+    passphrase: str = None,
+    payload: str = None,
+    dps_name: str = None,
+    id_scope: str = None,
+    resource_group_name: str = None,
+    login: str = None,
+    auth_type_dataplane: str = None,
+):
+    device_provider = DeviceRegistrationProvider(
```

Comment:
This seems like a lot of params needed just to initialize the DeviceRegistrationProvider.  Ideally initialization defines common params needed across provider methods. Then a provider method would take inputs specific for the operation.